delaymodel.py

```diff
@@ -234,55 +234,66 @@ def calculate_delay(self, publish : bool = True):
         makes use of the redis hash antenna coordinates and source coordinates. Therefore
         for custom coordinates, self.itrf or self.source must be overwritten.
         """
+        #initialise:
+        with self.source_points_lock:
+            sideband = self.source_pointings_dict['sideband']
+            sslo = self.source_pointings_dict['sslo']
+            self.source =  SkyCoord(self.source_pointings_dict['ra'], self.source_pointings_dict['dec'], unit='deg')
+
         while True:
             redis_publish_service_pulse(self.redis_obj, SERVICE_NAME)
 
             #Now, we want to check whether the source pointing has updated OR whether it has been over 5 seconds since
             #we last calculated delay coefficients:
-            t = time.time()
-            if ((t - self.delay_data["loadtime"]) >= 5 or self.source_point_update.is_set()):
-                #clear event here to avoid risk that the listening process sets the event as loadtime becomes > 5 and we miss a message.
+            t = time.time() * 1e6 #microseconds
+            if ((t - self.delay_data["loadtime_us"]) >= 5 or self.source_point_update.is_set()):
```

Comment:
This now tests if the load time is >5 usecs in the future

Reply:
Sigh, thanks good spot 🙄